asapdiscovery-modeling/asapdiscovery/modeling/schema.py

```diff
@@ -0,0 +1,80 @@
+from enum import Enum
+from pathlib import Path
+
+from asapdiscovery.data.schema import CrystalCompoundData, Dataset
+from pydantic import BaseModel, Field
+
+
+class MoleculeComponent(str, Enum):
+    PROTEIN = "protein"
+    LIGAND = "ligand"
+    WATER = "water"
+    OTHER = "other"
+
+
+class MoleculeFilter(BaseModel):
+    """Filter for selecting components of a molecule."""
+
+    protein_chains: list = Field(
+        list(),
+        description="List of chains containing the desired protein. An empty list will return all chains.",
+    )
+    ligand_chain: str = Field(
+        None,
+        description="Chain containing the desired ligand. An empty list will return all chains.",
+    )
+    water_chains: list = Field(
+        list(),
+        description="List of chains containing the desired water. An empty list will return all chains.",
+    )
+    other_chains: list = Field(
+        list(),
+        description="List of chains containing other items. An empty list will return all chains.",
+    )
+    components_to_keep: list[MoleculeComponent] = Field(
+        ["protein", "ligand", "water", "other"],
+        description="List of components to keep. An empty list will return all components.",
+    )
+
+
+class PrepOpts(BaseModel):
+    ref_fn: Path = Field(None, description="Reference structure to align to.")
+    ref_chain: str = Field(None, description="Chain ID to align to.")
+    seqres_yaml: Path = Field(None, description="Path to seqres yaml")
+    loop_db: Path = Field(None, description="Path to loop database to use for prepping")
+    output_dir: Path = Field(None, description="Directory to save output to.")
+
+
+class PreppedTarget(BaseModel):
```

Comment:
@apayne97 Thinking a bit about this over the weekend, adding schema kind of breaks the side-by-side idea, but we can add a different schema in a different place and then do a swap once we are happy, sound OK?

Reply:
Yeah i think we can consider this schema a stopgap to keep me from pulling my hair out anymore regarding the protein prep. I think basically I had been doing the 'side by side' approach for protein prep for a while and wanted to consolidate everything before the rest of the repo also undergoes a side by side approach.